datafusion/core/src/physical_plan/mod.rs

```diff
@@ -258,6 +258,32 @@ pub trait ExecutionPlan: Debug + Send + Sync {
     fn statistics(&self) -> Statistics;
 }
 
+/// Returns a new plan where all children were replaced by new plans if the provided children
+/// do not share the same point references with the existing children.
+/// The size of `children` must be equal to the size of `ExecutionPlan::children()`.
+/// Allow the vtable address comparisons for ExecutionPlan Trait Objects，it is harmless even
+/// in the case of 'false-native'.
```

Comment:
This should be false-positive I think?

Reply:
I think it should be 'false-native'(two expected equal things end up not comparing equal) for Trait objects.


[https://stackoverflow.com/questions/67109860/how-to-compare-trait-objects-within-an-arc]()

Reply:
Since we are reporting two equal items to be non-equal, it results in making a copy, so I term it 'false-positive'. 

It should be 'false negative' if you only mean data equality.  😆